File: S/OOP/C++/02_constructor_overloading.cpp
class Student 
{
    public:
        string name;
        int age;

        Student(): name("未知"), age(0) { }

        Student(string n): name(n), age(0) { }

        Student(string n, int a): name(n), age(a) { }

        // 委派建構子（C++11）：呼叫另一個建構子
        Student(string n) : Student(n, 0) { }
};

Student s1;
Student s2("A");
Student s3("B", 21);

File: S/OOP/Java/02_constructor_overloading.java
class Student {
    public String name;
    public int age;

    public Student() {
        this.name = "未知";
        this.age = 0;
    }

    public Student(String n) {
        this.name = n;
        this.age = 0;
    }

    public Student(String n, int a) {
        this.name = n;
        this.age = a;
    }

    public static void main(String[] args) {
        Student s1 = new Student();
        Student s2 = new Student("A");
        Student s3 = new Student("B", 21);
    }
}